substrate/primitives/weights/src/weight_meter.rs

```diff
@@ -149,6 +149,11 @@ impl WeightMeter {
 	pub fn can_consume(&self, w: Weight) -> bool {
 		self.consumed.checked_add(&w).map_or(false, |t| t.all_lte(self.limit))
 	}
+
+	/// Reclaim the given weight.
+	pub fn reclaim_proof_size(&mut self, s: u64) {
+		self.consumed.saturating_reduce(Weight::from_parts(0, s));
+	}
```

Comment:
I was thinking about this, however the `WeightMeter` was not offering any mechanism to reduce the consumed weight at all. I think because usually, once you have consumed some weight there is no way to get it back. I want to prevent users from even thinking about reclaiming ref time. The current API makes it very clear that you shall only ever reclaim proof size.